ambari-server/src/main/java/org/apache/ambari/server/serveraction/upgrades/OozieConfigCalculation.java

```diff
@@ -30,24 +31,103 @@
+
+  private static final String OOZIE_ENV_TARGET_CONFIG_TYPE = "oozie-env";
+  private static final String OOZIE_SITE_TARGET_CONFIG_TYPE = "oozie-site";
+  private static final String ELSERVICE_PROPERTIES_NAME_PREFIX = "oozie.service.ELService.ext.functions.";
   private static final String CONTENT_PROPERTY_NAME = "content";
+  private boolean oozie_env_updated = false;
+  private boolean oozie_site_updated = false;
 
   @Override
   public CommandReport execute(ConcurrentMap<String, Object> requestSharedDataContext)
     throws AmbariException, InterruptedException {
     String clusterName = getExecutionCommand().getClusterName();
     Cluster cluster = getClusters().getCluster(clusterName);
-    Config config = cluster.getDesiredConfigByType(TARGET_CONFIG_TYPE);
+    StringBuilder stdOutBuilder = new StringBuilder();
 
-    if (config == null) {
+    try {
+      changeOozieEnv(cluster, stdOutBuilder);
+    } catch (Exception e) {
       return  createCommandReport(0, HostRoleStatus.FAILED,"{}",
-                                   String.format("Source type %s not found", TARGET_CONFIG_TYPE), "");
+          String.format("Source type %s not found", OOZIE_ENV_TARGET_CONFIG_TYPE), "");
+    }
+
+    UpgradeContext upgradeContext = getUpgradeContext(cluster);
+    StackId targetStackId = upgradeContext.getTargetStack();
+
+
+    if (!serviceComponentSupport.isServiceSupported(FALCON_SERVICE_NAME, targetStackId.getStackName(), targetStackId.getStackVersion())) {
+      try {
+        removeFalconPropertiesFromOozieSize(cluster, stdOutBuilder);
+      } catch (AmbariException e) {
+        return createCommandReport(0, HostRoleStatus.FAILED, "{}",
+            String.format("Source type %s not found", OOZIE_SITE_TARGET_CONFIG_TYPE), "");
+      }
+    }
+
+    if (oozie_env_updated || oozie_site_updated) {
+      agentConfigsHolder.updateData(cluster.getClusterId(), cluster.getHosts().stream().map(Host::getHostId).collect(Collectors.toList()));
+    }
+
+    return createCommandReport(0, HostRoleStatus.COMPLETED, "{}",
+                  stdOutBuilder.toString(), "");
+  }
+
+  /**
+   * Changes oozie-site (removes oozie.service.ELService.ext.functions.*)
+   */
+  private void removeFalconPropertiesFromOozieSize(Cluster cluster, StringBuilder stringBuilder) throws AmbariException {
+    Config config = cluster.getDesiredConfigByType(OOZIE_SITE_TARGET_CONFIG_TYPE);
+
+    if (config == null) {
+      throw new AmbariException(String.format("Target config not found %s", OOZIE_SITE_TARGET_CONFIG_TYPE));
+    }
+
+    Map<String, String> properties = config.getProperties();
+    List<String> propertiesToRemove = properties.keySet().stream().filter(
+        s -> s.startsWith(ELSERVICE_PROPERTIES_NAME_PREFIX)).collect(Collectors.toList());
+
+    if (!CollectionUtils.isEmpty(propertiesToRemove)) {
+      stringBuilder.append(String.format("Removed following properties from %s: %s", OOZIE_SITE_TARGET_CONFIG_TYPE, propertiesToRemove));
```

Comment:
Do you need to use StringUtils.join() on the list of `propertiesToRemove` ?

Reply:
Currently the output looks like following : Removed following properties from oozie-site: [oozie.service.ELService.ext.functions.workflow, oozie.service.ELService.ext.functions.coord-action-create], I think it should be fine.

Reply:
Sure